File: app/src/main/java/com/example/nhakhoaapp/models/HoSoBenhAn.java
package com.example.nhakhoaapp.models;

import com.google.gson.annotations.SerializedName;

import java.util.Date;

public class HoSoBenhAn {
    @SerializedName("_id")
    private int id_ho_so;
    private int id_benhnhan;
    private int id_bac_si_kham;
    private Date ngay_kham;
    private String chan_doan;
    private String ket_qua_xet_nghiem;
    private String img;

    public HoSoBenhAn(int id_ho_so, int id_benhnhan, int id_bac_si_kham, Date ngay_kham, String chan_doan, String ket_qua_xet_nghiem, String img) {
        this.id_ho_so = id_ho_so;
        this.id_benhnhan = id_benhnhan;
        this.id_bac_si_kham = id_bac_si_kham;
        this.ngay_kham = ngay_kham;
        this.chan_doan = chan_doan;
        this.ket_qua_xet_nghiem = ket_qua_xet_nghiem;
        this.img = img;
    }
}

File: mongo-api/models/HoSoBenhAn.js
import mongoose from "mongoose";

const HoSoBenhAnSchema = new mongoose.Schema(
  {
    id_benh_nhan: {
      type: mongoose.Schema.Types.ObjectId,
      ref: "BenhNhan",
      required: true,
    },
    id_bac_si_kham: {
      type: mongoose.Schema.Types.ObjectId,
      ref: "NhanVien",
      required: true,
    },
    ngay_kham: { type: Date, required: true },
    chan_doan: { type: String },
    ket_qua_xet_nghiem: { type: String },
    img: { type: String },
  },
  { timestamps: true, collection: "hosobenhan" }
);

export default mongoose.model("HoSoBenhAn", HoSoBenhAnSchema);
